### ISSUE TITLE
Replace sprintf with snprintf

### ISSUE THREAD

**Issue by Hadatko** [STMicroelectronics/STMems_Standard_C_drivers#156]
### Device part numbers

-

### Summary

Brief explanation of the feature.

Hello, newer compilers are starting to complain related to usage of sprintf function (clang returning warning about deprecation). Their are suggesting to replace that call with snprintf. May you consider align your code?

### Motivation

```sh
'sprintf' is deprecated: This function is provided for compatibility reasons only. Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.
```

**Comment by avisconti** [comment]
Fixed by faefa8d

**Comment by Hadatko** [comment]
Funny i was on training from ST at that time 😂
